ai_assistant/relevancy.py

```diff
@@ -38,15 +40,53 @@ class RelevancyChecker:
     def __init__(self):
         self.model_id = os.environ.get("LIGHT_MODEL", DEFAULT_LIGHT_MODEL)
         self.client = genai.Client()
+        self.heroes: list[str] = []
+        self.items: list[str] = []
+        self._load_game_data()
+
+    def _load_game_data(self):
+        try:
+            self.heroes = list_heroes()
+            self.items = list_items()
+            LOGGER.info(f"Loaded {len(self.heroes)} heroes and {len(self.items)} items for relevancy checking")
+        except Exception as e:
+            LOGGER.error(f"Failed to load game data for relevancy checking: {e}")
+            self.heroes = []
+            self.items = []
+
+    def _check_keyword_match(self, prompt: str) -> bool:
+        prompt_lower = prompt.lower()
+
+        for hero in self.heroes:
+            if hero.lower() in prompt_lower:
+                LOGGER.info(f"Found hero match: '{hero}' in prompt")
+                return True
+
+        for item in self.items:
+            if item.lower() in prompt_lower:
+                LOGGER.info(f"Found item match: '{item}' in prompt")
+                return True
+
+        return False
 
     def is_relevant(self, prompt: str) -> bool:
+        if self._check_keyword_match(prompt):
+            return True
+
         try:
             if not os.environ.get("GEMINI_API_KEY"):
                 LOGGER.warning("GEMINI_API_KEY not set, will not run relevancy check")
                 return True
 
+            heroes_context = f"Known Heroes: {', '.join(self.heroes[:10])}" if self.heroes else ""
+            items_context = f"Known Items: {', '.join(self.items[:10])}" if self.items else ""
```

Comment:
If we slice it to 10 heroes we could still get problems, for example if I ask:
How many matches have I played on GT?
Won't give a keyword match, and if it's not in first 10 heroes, the light model can't infernthat gt is grey talon.